sample/sample.go

```diff
@@ -64,6 +145,24 @@ func Importance(samples, weights []float64, target dist.LogProber, proposal dist
 	}
 }
 
+// Rejectioner is a wrapper around the Rejection sampling generation procedure.
+// If the rejection sampling fails during the call to Sample, all samples will
+// be set to math.NaN().
+type Rejectioner struct {
```

Comment:
I thought about that, but it wasn't clear to me how it would be used. As of now, if my code takes in a Sampler, I wouldn't be able to check the Err field. If I know I'm doing rejection sample, wouldn't I call Rejection directly?